getpaid/backends/pagseguro/__init__.py

```diff
@@ -0,0 +1,204 @@
+        "token":""
+    }
+
+    _PRODUCT_DATA = {
+        "currency": "BRL",
+        "itemId1":"",
+        "itemDescription1":"",
+        "itemQuantity1":"1",
+        "itemAmount1":"",
+        "reference":"",
+    }
+
+
+    def get_gateway_url(self, request):
+        """  """
+        payp = PaymentProcessor
+        gateway_url = payp.get_backend_setting('gateway_url_start_transac',
+                                            self._GATEWAY_URL_START_TRANSAC)
+        gateway_url2 = payp.get_backend_setting('gateway_url_payment', 
+                                                self._GATEWAY_URL_PAYMENT)
+        
+        is_dev = payp.get_backend_setting('testing', False)
+        if is_dev:
+            gateway_url = payp.get_backend_setting('dev_gateway_url_start_transac',
+                                            self._DEV_GATEWAY_URL_START_TRANSAC)
+            gateway_url2 = payp.get_backend_setting('dev_gateway_url_payment', 
+                                                self._DEV_GATEWAY_URL_PAYMENT)
+
+        order = self.payment.order
+
+        token = PaymentProcessor.get_backend_setting('token')
+        email = PaymentProcessor.get_backend_setting('email')
+        self._ACOUNT_DATA["token"] = token
+        self._ACOUNT_DATA["email"] = email
+
+        reference = uuid.uuid4().hex
+
+        product_description = self.get_order_description(self.payment, self.payment.order)
+        redirectURL = PaymentProcessor._get_view_full_url(request, 'getpaid-pagseguro-success', args=(self.payment.id,))
+        
+        self._PRODUCT_DATA.update(itemId1=self.payment.id,
+                                  itemDescription1= product_description.encode("latin1", "ignore"), 
+                                  currency=self.payment.currency,
+                                  itemAmount1=self.payment.amount,
+                                  reference=reference,
+                                  redirectURL=redirectURL,
+                                  )
+
+        full_data = dict(self._ACOUNT_DATA, **self._PRODUCT_DATA)
+        params = urlencode(full_data)
+
+        payment_full_url = "%s%s" % (gateway_url, params)
+        request.encoding = 'ISO-8859-1'
+        
+        logger.info(payment_full_url)
+
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        response = requests.post(payment_full_url, headers=headers).text
+
+        code = ""
+        logger.info("xml response from pagseguro: " + str(response.encode("utf-8")))
+        dom = parseString(response)
+        
+        checkout = dom.getElementsByTagName("checkout")
+        if checkout:
+            childNodes = checkout[0].childNodes
+            code = childNodes[0].firstChild.nodeValue
+            date = childNodes[1].firstChild.nodeValue
+
+        self.payment.external_id = reference
+        self.payment.description = code
+        self.payment.save();
+        
+        return "%s%s" % (gateway_url2, code), 'GET', {}
```

Comment:
I hope I'm not too paranoid but could you use `urlencode` with all params, especially if they come from external site like `code` here?